src/main/java/net/openhft/chronicle/queue/impl/single/StoreAppender.java

```diff
@@ -91,16 +90,20 @@ class StoreAppender extends AbstractCloseable
 
         try {
             queue.cleanupStoreFilesWithNoData();
+            int lastExistingCycle = queue.lastCycle();
+            // lastCycle() can be negative (no files on disk) or a cycle number of last existing cycle file
+            // Initial cycle should be assigned to last existing cycle file unless it is in the future
+            // If last existing file is in the past, it's still possible it will be replicated/backfilled so no EOF
+            cycle = Math.min(queue.cycle(), lastExistingCycle >= 0 ? lastExistingCycle : queue.cycle());
```

Comment:
Minor, but it feels like this would be more readable if it were
```
cycle = lastExistingCycle >= 0 ? Math.min(lastExistingCycle, queue.cycle()) : queue.cycle()
```
Just to make it clear that if `lastExistingCycle < 0` we don't need to do the `Math.min()`

Or even flip the ternary, just move it outside the call to `Math.min`

Reply:
Seems silly to do have a path that does `Math.min(queue.cycle(), queue.cycle())`

Reply:
I've simplified that condition